tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcCommon.cs

```diff
@@ -0,0 +1,119 @@
+﻿// <copyright file="GrpcCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Grpc
+{
+    internal static class GrpcCommon
+    {
+        public const string OperationName = "grpc.request";
+        public const string ServiceName = "grpc-client";
+        public const string RequestMetadataTagPrefix = "grpc.request.metadata";
+        public const string ResponseMetadataTagPrefix = "grpc.response.metadata";
+
+        public static void AddGrpcTags(GrpcTags tags, Tracer tracer, int grpcType, string? name, string? path, string? serviceName)
+            => AddGrpcTags(tags, tracer, GetGrpcMethodKind(grpcType), name, path, serviceName);
+
+        public static void AddGrpcTags(GrpcTags tags, Tracer tracer, string grpcType, string? name, string? path, string? serviceName)
+        {
+            tags.MethodKind = grpcType;
+            tags.MethodName = name;
+            tags.MethodPath = path;
+
+            if (!string.IsNullOrEmpty(serviceName))
+            {
+                // get the package and service name
+                var indexOf = serviceName!.LastIndexOf('.');
+
+                if (indexOf > 0 && indexOf < serviceName.Length - 2)
+                {
+                    tags.MethodPackage = serviceName.Substring(startIndex: 0, length: indexOf);
+                    tags.MethodService = serviceName.Substring(startIndex: indexOf + 1, length: serviceName.Length - indexOf - 1);
+                }
+                else
+                {
+                    tags.MethodService = serviceName;
+                }
+            }
+
+            tags.SetAnalyticsSampleRate(IntegrationId.Grpc, tracer.Settings, enabledWithGlobalSetting: false);
+        }
+
+        public static string GetGrpcMethodKind(int value)
+            => value switch
+            {
+                // Values from Grpc.Core.MethodType
+                0 => GrpcMethodKinds.Unary,
+                1 => GrpcMethodKinds.ClientStreaming,
+                2 => GrpcMethodKinds.ServerStreaming,
+                3 => GrpcMethodKinds.DuplexStreaming,
+                _ => GrpcMethodKinds.Unary, // No other values are valid, but probably don't want to throw here?
```

Comment:
This feels like it could lead to subtle errors if the enum every did change (although, that seems fairly unlike).

It's perhaps better to cast value to the enum type when there's no known value:

```suggestion
                _ => (GrpcMethodKinds)value, // No other values are valid, but probably don't want to throw here?
```

Reply:
Switched to `"Unknown"` in https://github.com/DataDog/dd-trace-dotnet/pull/2535/commits/49ace7ef4457ed9607e66de8aba0b869d369a020